### PR TITLE
NEOS-14978: add check to skip setting transformer in apply default transformer if…

### PR DESCRIPTION
**Pull request by evisdrenova**
Upgrades the Apply Default Transformer to be a little smarter and give the user more options.  Before it was all or nothing to apply the defaults if you clicked it. Now if there are any transformers that are set, it will prompt you and ask you if you want to overwrite those. As a user you can decide to overwrite what you have already set or just set everything else around it.

https://www.loom.com/share/107c9c351efb4fcb81b1f15c99b7d0bd?sid=f75b1f53-30e0-41b8-83ba-549b49d709a5

### CODE REVIEW

**Review by nickzelei — CHANGES_REQUESTED** [review]
left a few comments to clean up a bit!

File: frontend/apps/web/components/jobs/SchemaTable/SchemaTableToolBar.tsx [hunk 1]
```diff
@@ -172,49 +210,57 @@ export function SchemaTableToolbar<TData>({
                 </Button>
               }
               headerText="Apply Default Transformers?"
-              description="This setting will apply the 'Passthrough' Transformer to every column that is not Generated, while applying the 'Use Column Default' Transformer to all Generated (non-Identity) columns."
-              buttonText="Apply"
+              description={handleAlertDescriptionBody()}
```
[the remark this answers is not in the document]
Comment (evisdrenova) [reply frontend/apps/web/components/jobs/SchemaTable/SchemaTableToolBar.tsx:213]: Yeah I hear you - my concern with that was that it wouldn't be obvious to the user if they had already mapped transformers or not if we just put a switch that said "Enable to overwrite your Transformers". That doesn't tell the user that they have already selected Transfomers. I couldn't find anything in the react hook form api that already took care of this. They have `isDirty` but that doesn't give us what we want. 

But agree on the scalability of it. If we're okay with not making it obvious to the use if they've already mapped transformers then I can implement the form with the optional switch and update the logic.

[the remark this answers is not in the document]
Comment (evisdrenova) [reply frontend/apps/web/components/jobs/SchemaTable/SchemaTableToolBar.tsx:213]: updated

File: frontend/apps/web/yup-validations/jobs.ts [hunk 2]
```diff
@@ -274,3 +274,11 @@ export const DataSyncSourceFormValues = SourceFormValues.concat(
 export type DataSyncSourceFormValues = Yup.InferType<
   typeof DataSyncSourceFormValues
 >;
+
+export const DEFAULT_TRANSFORMER_FORM_VALUES = Yup.object({
```
Comment (nickzelei) [inline frontend/apps/web/yup-validations/jobs.ts:278]: Moving away from having this and the type be different. This value should be the same casing as the type so they are intertwined. 
```suggestion
export const DefaultTransformerFormValues = Yup.object({
```

**Review by nickzelei — APPROVED** [review]

**Pull request merged**